Base/splitter.h

```diff
@@ -0,0 +1,42 @@
+#ifndef SPLITTER_H__
+#define SPLITTER_H__
+
```

Comment:
Namespace would follow this pattern: {project_name}::{folders}*. In the case of the splitter.h placed in the folder tree inside the root folder, we would have the following namespace machine_learning::tree, is that right?

I am not using any namespace in this commit. Let we decide first, and then we refactor the code with the proper namespaces.